File: Lab2/app/src/main/java/com/example/firstjavaapp/KinoPoiskService.java
package com.example.firstjavaapp;
import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;


public class KinoPoiskService {
    private static KinoPoiskService mInstance;
    private static final String BASE_URL = "https://api.kinopoisk.dev";
    private final Retrofit mRetrofit;

    private KinoPoiskService() {
        mRetrofit = new Retrofit.Builder()
                .baseUrl(BASE_URL)
                .addConverterFactory(GsonConverterFactory.create())
                .build();
    }

    public static KinoPoiskService getInstance() {
        if (mInstance == null) {
            mInstance = new KinoPoiskService();
        }
        return mInstance;
    }

    public KinoPoiskApi getJSONApi() {
        return mRetrofit.create(KinoPoiskApi.class);
    }
}

File: Lab_2_NEW/app/src/main/java/com/example/lab_2_new/KinoPoiskService.kt
package com.example.lab_2_new

import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory


class KinoPoiskService {

    private val BASE_URL = "https://api.kinopoisk.dev"
    private var mRetrofit: Retrofit? = null

    init {
        mRetrofit = Retrofit.Builder()
            .baseUrl(BASE_URL)
            .addConverterFactory(GsonConverterFactory.create())
            .build()
    }

    companion object{
        private var mInstance: KinoPoiskService? = null
        fun getInstance(): KinoPoiskService? {
            if (mInstance == null) mInstance = KinoPoiskService()
            return mInstance
        }
    }

    fun getJSONApi(): KinoPoiskApi? {
        return mRetrofit!!.create(KinoPoiskApi::class.java)
    }
}